Sources/MongoSwift/BSON/BsonValue.swift

```diff
@@ -460,10 +469,16 @@ public struct CodeWithScope: BsonValue {
         let scopeDoc = Document(fromPointer: scopeData)
         return CodeWithScope(code: code, scope: scopeDoc)
     }
+
+    public static func == (lhs: CodeWithScope, rhs: CodeWithScope) -> Bool {
+        return lhs.code == rhs.code && lhs.scope == rhs.scope
+    }
 }
 
 /// A struct to represent the BSON MaxKey type.
-public struct MaxKey: BsonValue, Equatable {
+public struct MaxKey: BsonValue, Equatable, Decodable {
+    private var maxKey = 1
```

Comment:
I added this private var to deal with the fact that anything was parseable as a MinKey or MaxKey